packages/bolt-connection/test/connection-provider/connection-provider-routing.test.js

```diff
@@ -2694,13 +2783,49 @@ describe('#unit RoutingConnectionProvider', () => {
           pool
         )
 
+        const serverInfo = connectionProvider = await connectionProvider.verifyConnectivityAndGetServerInfo({ database, accessMode })
+        expect(serverInfo).toEqual(new ServerInfo({}, 4.4))
+      })
+    })
+
+    describe('when there is not server available in the routing table', () => {
+      it('should thown an discovery error', async () => {
+        const expectedError = newError(
+          `No servers available for database '${database || null}' with access mode '${accessMode || READ}'`,
+          SERVICE_UNAVAILABLE
+        )
+        const routingTable = newRoutingTable(
+          database || null,
+          [server1, server2],
+          accessMode === READ ? [] : [server3, server4],
+          accessMode === WRITE ? [] : [server5, server6],
+          Integer.MAX_SAFE_VALUE
+        )
+
+        const routingTableToRouters = {}
+        const routingMap = {}
+        routingMap[server0.asKey()] = routingTable
+        routingMap[server1.asKey()] = routingTable
+        routingMap[server2.asKey()] = routingTable
+        routingTableToRouters[database || null] = routingMap
+
+        const connectionProvider = newRoutingConnectionProviderWithSeedRouter(
+          server0,
+          [server0], // seed router address resolves just to itself
+          [
+            routingTable
+          ],
+          routingTableToRouters
+        )
+
         try {
-          connectionProvider = await connectionProvider.verifyConnectivityAndGetServerInfo({ database, accessMode })
+          await connectionProvider.verifyConnectivityAndGetServerInfo({ database, accessMode })
           expect().toBe('not reached')
-        } catch (e) {
-          expect(e).toBe(error)
+        } catch (error) {
+          expect(error).toEqual(expectedError)
         }
       })
+
```

Comment:
:eyes:

Reply:
Typo and extra empty line are my signature.